masonry/src/widget/board.rs

```diff
@@ -0,0 +1,464 @@
+        self.ctx.get_mut(&mut self.widget.children[idx])
+    }
+
+    pub fn clear(&mut self) {
+        if !self.widget.children.is_empty() {
+            self.ctx.request_layout();
+
+            for child in self.widget.children.drain(..) {
+                self.ctx.remove_child(child);
+            }
+        }
+    }
+}
+
+impl<'a, W: Widget> WidgetMut<'a, PositionedElement<W>> {
+    pub fn inner_mut(&mut self) -> WidgetMut<'_, W> {
+        WidgetMut {
+            ctx: self.ctx.reborrow_mut(),
+            widget: &mut self.widget.inner,
+        }
+    }
+}
+
+impl<'a> WidgetMut<'a, Box<dyn SvgElement>> {
+    /// Attempt to downcast to `WidgetMut` of concrete Widget type.
+    pub fn try_downcast<W2: Widget>(&mut self) -> Option<WidgetMut<'_, W2>> {
+        Some(WidgetMut {
+            ctx: self.ctx.reborrow_mut(),
+            widget: self.widget.as_mut_any().downcast_mut()?,
+        })
+    }
+
+    /// Downcasts to `WidgetMut` of concrete Widget type.
+    ///
+    /// ## Panics
+    ///
+    /// Panics if the downcast fails, with an error message that shows the
+    /// discrepancy between the expected and actual types.
+    pub fn downcast<W2: Widget>(&mut self) -> WidgetMut<'_, W2> {
+        let w1_name = self.widget.type_name();
+        match self.widget.as_mut_any().downcast_mut() {
+            Some(widget) => WidgetMut {
+                ctx: self.ctx.reborrow_mut(),
+                widget,
+            },
+            None => {
+                panic!(
+                    "failed to downcast widget: expected widget of type `{}`, found `{}`",
+                    std::any::type_name::<W2>(),
+                    w1_name,
+                );
+            }
+        }
+    }
+}
+
+// --- MARK: IMPL WIDGET BOARD ---
+impl Widget for Board {
+    fn on_pointer_event(&mut self, _ctx: &mut EventCtx, _event: &PointerEvent) {}
+    fn on_text_event(&mut self, _ctx: &mut EventCtx, _event: &TextEvent) {}
+    fn on_access_event(&mut self, _ctx: &mut EventCtx, _event: &AccessEvent) {}
+    fn on_status_change(&mut self, _ctx: &mut LifeCycleCtx, _event: &StatusChange) {}
+
+    fn lifecycle(&mut self, ctx: &mut LifeCycleCtx, event: &LifeCycle) {
+        for child in &mut self.children {
+            child.lifecycle(ctx, event);
+        }
+    }
+
+    fn layout(&mut self, ctx: &mut LayoutCtx, bc: &BoxConstraints) -> Size {
+        bc.debug_check("Board");
+
+        for child in &mut self.children {
+            let (size, origin) = {
+                let child_ref = ctx.get_raw_ref(child);
+                (child_ref.widget().size(), child_ref.widget().origin())
+            };
+            ctx.run_layout(child, &BoxConstraints::tight(size));
+            ctx.place_child(child, origin);
+        }
+
+        bc.max()
+    }
+
+    fn paint(&mut self, _ctx: &mut PaintCtx, _scene: &mut Scene) {}
+
+    fn accessibility_role(&self) -> Role {
+        Role::GenericContainer
+    }
```

Comment:
Open question: is this the right role for this widget, or is there a more specific role for it? I guess "Canvas" doesn't really apply.